ruby/lib/simple_inline_text_annotation/parser.rb

```diff
@@ -40,25 +42,54 @@ def get_obj_for(label)
       @entity_type_collection.get(label) || label
     end
 
-    def process_denotations(full_text)
-      while full_text =~ DENOTATION_PATTERN
-        match = Regexp.last_match
-        process_single_denotation(match, full_text)
+    def process_annotations(full_text)
+      current_pos = 0
+
+      while (match = ANNOTATION_PATTERN.match(full_text, current_pos))
+        current_pos = process_annotation_and_update_position(match, full_text)
       end
     end
 
-    def process_single_denotation(match, full_text)
+    def process_annotation_and_update_position(match, full_text)
       target_text = match[1]
-      label = match[2]
-
       begin_pos = match.begin(0)
       end_pos = begin_pos + target_text.length
-      obj = get_obj_for(label)
 
-      @denotations << Denotation.new(begin_pos, end_pos, obj)
+      return match.end(0) unless process_annotation_by_size(match[2], begin_pos, end_pos)
 
-      # Replace the processed annotation with its text content
       full_text[match.begin(0)...match.end(0)] = target_text
+      match.begin(0) + match[1].length
```

Comment:
```suggestion
      end_pos
```

既に計算している値があるので、それを返すのが良さそうです。
また、「置き換えた後の文字列の終わり」という意図も伝わりやすくなります。

Reply:
返り値を修正しています。

[Modified the return value of process_annotation_and_update_position](https://github.com/Tamada-Arino/simple-inline-text-annotation/pull/29/commits/86e53fe17957b45c8c2d0c6877ea31afa71d589c)